### PR TITLE
feature: support queueing_hint_execution_duration_seconds metric

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

support `queueing_hint_execution_duration_seconds` metric.
It's required for visibility of the bottleneck more from `event_handling_duration_seconds`, which specific QHint(s) take too much time.

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/kubernetes/issues/124566

#### Does this PR introduce a user-facing change?

```release-note
implement `queueing_hint_execution_duration_seconds` metric, which is the time the QueueingHint function takes.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sanposhiho** [comment]
/hold

to go thru two reviews.

/cc @gabesaba @macsko @alculquicondor

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 1]
```diff
@@ -509,6 +510,8 @@ func (p *PriorityQueue) isPodWorthRequeuing(logger klog.Logger, pInfo *framework
 				}
 				hint = framework.Queue
 			}
+			metrics.QueueingHintExecutionDuration.WithLabelValues(hintfn.PluginName, event.Label, queueingHintToLabel(hint, err)).Observe(metrics.SinceInSeconds(start))
```
Comment (alculquicondor) [inline pkg/scheduler/internal/queue/scheduling_queue.go:513]: Just recording this metric has the potential of increasing latency, especially for such a small operation (ideally).

Look what we did for PluginExecutionDuration https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/metrics/metric_recorder.go#L121

Maybe we need to do something similar. It basically discards the metric update if we can't keep up with flushing.

Comment (sanposhiho) [reply pkg/scheduler/internal/queue/scheduling_queue.go]: Done.

File: pkg/scheduler/metrics/metric_recorder.go [hunk 2]
```diff
@@ -119,9 +119,19 @@ func NewMetricsAsyncRecorder(bufferSize int, interval time.Duration, stopCh <-ch
 // ObservePluginDurationAsync observes the plugin_execution_duration_seconds metric.
 // The metric will be flushed to Prometheus asynchronously.
 func (r *MetricAsyncRecorder) ObservePluginDurationAsync(extensionPoint, pluginName, status string, value float64) {
+	r.observeMetricAsync(PluginExecutionDuration, value, pluginName, extensionPoint, status)
+}
+
+// ObservePluginDurationAsync observes the plugin_execution_duration_seconds metric.
```
Comment (macsko) [inline pkg/scheduler/metrics/metric_recorder.go:125]:
```suggestion
// ObserveQueueingHintDurationAsync observes the queueing_hint_execution_duration_seconds metric.
```

**Comment by sanposhiho** [comment]
/retest
/assign @kerthcet 
for approval

File: pkg/scheduler/metrics/metrics.go [hunk 3]
```diff
@@ -198,6 +206,19 @@ var (
 		},
 		[]string{"plugin", "extension_point", "status"})
 
+	// This is only available when the QHint feature gate is enabled.
+	queueingHintExecutionDuration = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Subsystem: SchedulerSubsystem,
+			Name:      "queueing_hint_execution_duration_seconds",
+			Help:      "Duration for running a queueing hint from a plugin.",
```
Comment (kerthcet) [inline pkg/scheduler/metrics/metrics.go:214]:
```suggestion
			Help:      "Duration for running a queueing hint function of a plugin.",
```

Comment (sanposhiho) [reply pkg/scheduler/metrics/metrics.go]: @kerthcet Done.

**Pull request merged**